android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/GleanUtil.kt

```diff
@@ -0,0 +1,134 @@
+package org.mozilla.firefox.vpn.daemon
+
+import android.content.Context
+import android.util.Log
+import mozilla.components.service.glean.BuildInfo
+import mozilla.components.service.glean.Glean
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.service.glean.private.EventExtras
+import mozilla.telemetry.glean.net.HttpURLConnectionUploader
+import mozilla.telemetry.glean.private.EventMetricType
+import mozilla.telemetry.glean.private.NoExtraKeys
+import mozilla.telemetry.glean.private.NoExtras
+import org.json.JSONObject
+import org.mozilla.firefox.vpn.daemon.GleanMetrics.Pings
+import org.mozilla.firefox.vpn.daemon.GleanMetrics.Sample
+import java.util.*
+import kotlin.reflect.KParameter
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.primaryConstructor
+
+// @MainThread
+@Suppress("UNUSED") // All the calls are via jni from the client, so we can ignore
+class GleanUtil(aParent: Context) {
+    private val mParent: Context = aParent
+
+    fun initializeGlean() {
+        val gleanEnabled = Prefs.get(mParent).getBoolean("glean_enabled", false) // Don't send telemetry unless explicitly asked to.
+        val channel = if (mParent.packageName.endsWith(".debug")) {
+            "staging"
+        } else {
+            "production"
+        }
+        val conf = Configuration(HttpURLConnectionUploader(), Configuration.DEFAULT_TELEMETRY_ENDPOINT, channel, null)
+        val build = BuildInfo("versionCode", "VersionName", Calendar.getInstance())
+        Glean.registerPings(Pings)
+        Log.e("ANDROID-GLEAN", "initializeGlean on:$gleanEnabled, ch:$channel ")
+        Glean.initialize(
+            applicationContext = mParent.applicationContext,
+            uploadEnabled = gleanEnabled,
+            buildInfo = build,
+            configuration = conf
+        )
+        Log.i("Android-Glean", "Initialised android glean")
+    }
+
+    fun setGleanUploadEnabled(upload: Boolean) {
+        Log.e("ANDROID-GLEAN", "setGleanUploadEnabled  $upload")
+        Prefs.get(mParent).edit().apply {
+            putBoolean("glean_enabled", upload)
+            apply()
+        }
+        Glean.setUploadEnabled(upload)
+    }
+
+    fun setGleanSourceTag(tags: String = "") {
+        // val tag_list = tag.split(",")
+        // Glean.setSourceTags(tag_list);
+        // TODO: Glean-Android does not support this, yet.

```

Comment:
This is now supported from [v44.2.0](https://github.com/mozilla/glean/blob/main/CHANGELOG.md#v4420-2022-05-16) on.

Reply:
Nice! Thank you, so I guess we need to wait for android-components to pick that new version up. - 
https://github.com/mozilla-mobile/android-components/blob/main/buildSrc/src/main/java/Dependencies.kt#L35

Reply:
/o\ Yeap, I guess we do.

Reply:
Right. That requires us updating it in m-c, however we're preparing a major update for next week (v50) and so I won't get to a 44.2.0 update there.
Are you already depending on other Android Components? If not then you can depend on the Glean release directly using `org.mozilla.telemetry:glean:44.2.0`. That way you don't need to wait for the (sometimes lengthy) a-c release cycles.